lib/Doctrine/ORM/Persisters/Collection/ManyToManyPersister.php

```diff
@@ -757,4 +758,13 @@ private function getLimitSql(Criteria $criteria): string
 
         return $this->platform->modifyLimitQuery('', $limit, $offset ?? 0);
     }
+
+    private function getMapping(PersistentCollection $collection): AssociationMapping&ManyToManyAssociationMapping
+    {
+        $mapping = $collection->getMapping();
+
+        assert($mapping instanceof ManyToManyAssociationMapping);
```

Comment:
Is this assert even needed here when the type hint only allows `AssociationMapping` instances that implement `ManyToManyAssociationMapping`?

Reply:
Yes, otherwise SA will complain that nothing proves we return that type although we claim we do.